core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -169,13 +169,14 @@ private static JSONObject readInputJson(Context context, int resourceId) throws
     public CategoryConfiguration forCategoryType(@NonNull CategoryType categoryType) throws AmplifyException {
         final CategoryConfiguration categoryConfiguration =
             categoryConfigurations.get(categoryType.getConfigurationKey());
-        if (categoryConfiguration == null) {
-            throw new AmplifyException(
-                "Unknown/bad category type: " + categoryType,
-                "Be sure to use one of the supported Categories in your current version of Amplify"
-            );
-        }
-        return categoryConfiguration;
+        return categoryConfiguration != null ? categoryConfiguration :
+            new CategoryConfiguration() {
```

Comment:
If I have a config file like:
```
{
    "foo": ...
}
```
I believe the current behavior is that I'll get a non-null config for `FooCategory`, but I'd get a `null` config for `BarCategory`, right? I do actually thing that's a correct use of optionality in Java. The JSON document does not contain/specify any config for `"bar"`.

Is there anyway we can preserve that attribute of the system?

If not, as a mechnical note to simplify the current code block, I'd suggest pulling this anonymous `CategoryConfiguration` out into a subclass. You could make like a :
```
final class EmptyCategoryConfiguration extends CategoryConfiguration {
    private final CategoryType categoryType;

    private EmptyCategoryConfiguration(CategoryType categoryType) {
        this.categoryType = categoryType;
    }

    static EmptyCategoryConfiguration forCategory(CategoryType categoryType) {
        Objects.requireNonNull(categoryType);
        return new EmptyCategoryConfiguration(categoryType);
    }
    
    @NonNull
    @Override
    public CategoryType getCategoryType() {
        return categoryType;
    }
}
```
```
EmptyCategoryConfiguration.forCategory(CategoryType.DATASTORE); // Java is verbose isn't it, gee.
```
Hm. Double check the source and test code for current extensions of `CategoryConfiguration` though - I feel like something like this might already be there? (I don't remember.) It probably has a different name, if it exists. You could "promote" it from test to source code and fixup the name, if its around.